src/ImageSharp.Drawing/Processing/PathGradientBrush.cs

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.Primitives;
+using SixLabors.Shapes;
+
+namespace SixLabors.ImageSharp.Processing
+{
+    /// <summary>
+    /// Provides an implementation of a brush for painting gradients between multiple color positions in 2D coordinates.
+    /// It works similarly with the class in System.Drawing.Drawing2D of the same name.
+    /// </summary>
+    public sealed class PathGradientBrush : IBrush
+    {
+        private readonly Polygon path;
+
+        private readonly IList<Edge> edges;
+
+        private readonly Color centerColor;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PathGradientBrush"/> class.
+        /// </summary>
+        /// <param name="lines">Line segments of a polygon that represents the gradient area.</param>
+        /// <param name="colors">Array of colors that correspond to each point in the polygon.</param>
+        /// <param name="centerColor">Color at the center of the gradient area to which the other colors converge.</param>
+        public PathGradientBrush(ILineSegment[] lines, Color[] colors, Color centerColor)
+        {
```

Comment:
We need arg. check here:
`lines`, `colors`: not null, and counts are matching

Reply:
Sure, I considered adding them but thought it wasn't a convention after looking at other brush implementations (see my first comment).

Would it be ok if I just throw `ArgumentNullException` / `ArgumentOutOfRangeException` as I see fit? And, should I also add associated tests?

P. S.: I'm a Godot developer by the way, so yeah I really should be aware of such things :)

Reply:
It's done. Please let me know if you want me to change anything else.